### PR TITLE
Feature/randomizer

### PR DESCRIPTION
**Pull request by CBond05**

### CODE REVIEW

File: src/components/App/Randomizer/Randomizer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            return;
+        }
+
+        fireEvent.change(textarea, { target: { value: items } });
+        fireEvent.change(input, { target: { value: 2 } });
+        fireEvent.click(button);
+        const li = baseElement.querySelectorAll('li');
+        expect(li).toHaveLength(2);
+    });
+});
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.test.tsx]: All tests Ctrl + c -> Ctrl + v from Signup.test.tsx?

File: src/components/App/Randomizer/ListItem.css [hunk 2]
```diff
@@ -0,0 +1,46 @@
+.MuiInput-root {
+    position: relative;
+    height: 48px;
+    width: 50%;
+}
+@media screen and (min-width: 720px) {
+    #outlined-multiline-static {
```
Comment (letrylesha) [inline src/components/App/Randomizer/ListItem.css]: ID for stylind is bad, use class

File: src/components/App/Randomizer/ListItem.css [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+@media screen and (min-width: 720px) {
+    #outlined-multiline-static2 {
+        display: flex;
+        position: relative;
+        width: 100%;
+        border-radius: 1px;
+        margin-bottom: 5px;
+    }
+}
+ #RandomizeButton{
```
Comment (letrylesha) [inline src/components/App/Randomizer/ListItem.css]: leave camelcase for variables in js, write like random-button

File: src/components/App/Randomizer/Randomizer.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import Button from '@material-ui/core/Button';
+import Input from '@material-ui/core/Input';
+import './ListItem.css'
+import {StyledTextArea, StyledRandom, AreaWrapper, NavWrapper, ResultArea, TheList} from './StyledRandomizerComponents'
+import {ResultInterface} from './ResultItemInterface'
+
+export default function Randomizer() {
+  const [value, setValue] = React.useState('Your result will be here');
+  const [result, setResult] = React.useState<ResultInterface[]>([]);
+  const [inputValue, setInputValue]= React.useState(1);
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: Just useState

File: src/components/App/Randomizer/Randomizer.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  };
+
+  let resultArray: string[] = [];
+  const randomize = () => {
+    resultArray = value.split('\n');
+    if(resultArray.length>=inputValue){
+      let res = chooseRandom(resultArray);
+      setResult([...res]);
+    } else {
+      setResult([{id: 0, value: "Incorrect input."}]);
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: Wtf? create initial state

File: src/components/App/Randomizer/Randomizer.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    }
+  }
+  
+  const chooseRandom = (resultArray: string[]) =>
+  {
+    let currentIndex = resultArray.length, 
+    temporaryValue, randomIndex, resultIndex, arr: ResultInterface[]=[], loopCounter=0;
+    setResult([]);
+    let filteredArray = [];
+    let uniqueValues = new Set(resultArray);
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: let, let is everywhere. If you're not going to change this value use const

File: src/components/App/Randomizer/Randomizer.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+               resultArray[currentIndex] = resultArray[randomIndex];
+               resultArray[randomIndex] = temporaryValue;
+             }
+           resultIndex = Math.floor(Math.random() * resultArray.length);
+           let ItemOfResult = {id: 0, value: ""};
+           ItemOfResult.value = resultArray[resultIndex];
+           ItemOfResult.id = Math.random()*1e8;
+            let isExists = arr.some((item) =>{
+              return item.value === ItemOfResult.value
+            })
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: write in one line
arr.some((item) => item.value === ItemOfResult.value )

File: src/components/App/Randomizer/Randomizer.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+          <Input id="RandomizeInput" type="number" color="primary" onChange={handleInputChange} placeholder="Input number of results. Default value is 1"></Input>
+        </NavWrapper> 
+        <StyledTextArea onChange={(e: any) => handleChange(e)}>
+        </StyledTextArea>
+        <ResultArea>
+          <TheList>
+            {result.map((item) => {
+            return <li key={item.id}>{item.value}</li>     
+            })
+            }
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: also you can write it in one line

File: src/components/App/Randomizer/Randomizer.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import Input from '@material-ui/core/Input';
+import './ListItem.css'
+import {StyledTextArea, StyledRandom, AreaWrapper, NavWrapper, ResultArea, TheList} from './StyledRandomizerComponents'
+import {ResultInterface} from './ResultItemInterface'
+
+export default function Randomizer() {
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: We don't use this style in our project. Use arrow function and export your component in bottom (I mean after your code)

File: src/components/App/Randomizer/Randomizer.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+           if(!isExists && ItemOfResult.value !== ""){
+              arr.push(ItemOfResult);
+           } else{
+             loopCounter--;
+           }
+             loopCounter++;
+          }
+        return arr;
+    }
+  }
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.tsx]: From my persective we can split this code by functions (whole while loop, open code)

File: src/components/App/Randomizer/Randomizer.test.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import Randomizer from './Randomizer';
+
+describe('Randomizer test', () => {
+    let component = ()=> {
```
Comment (letrylesha) [inline src/components/App/Randomizer/Randomizer.test.tsx]: where is afterEach(cleanup); ?

**Review by letrylesha — CHANGES_REQUESTED** [review]

File: src/components/App/Randomizer/Randomizer.test.tsx (hunk 1 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.test.tsx]: old version, forgot to replace with .ToBeInTheDocument(). Now changed
All on my own

File: src/components/App/Randomizer/ListItem.css (hunk 2 above)
Comment (CBond05) [reply src/components/App/Randomizer/ListItem.css]: deprecated selector, forgot to delete

File: src/components/App/Randomizer/ListItem.css (hunk 3 above)
Comment (CBond05) [reply src/components/App/Randomizer/ListItem.css]: changed
ID's are used in order to rewrite MaterialUi styles

File: src/components/App/Randomizer/Randomizer.tsx (hunk 4 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: default values setup

File: src/components/App/Randomizer/Randomizer.tsx (hunk 5 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: initial state is set above. This is set such way in order to handle exeptions

File: src/components/App/Randomizer/Randomizer.tsx (hunk 6 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: changed

File: src/components/App/Randomizer/Randomizer.tsx (hunk 7 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: changed

File: src/components/App/Randomizer/Randomizer.tsx (hunk 8 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: changed

File: src/components/App/Randomizer/Randomizer.tsx (hunk 9 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: changed

File: src/components/App/Randomizer/Randomizer.tsx (hunk 10 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: changed

File: src/components/App/Randomizer/Randomizer.test.tsx (hunk 11 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.test.tsx]: added

File: src/components/App/Randomizer/StyledRandomizerComponents.tsx [hunk 12]
```diff
@@ -0,0 +1,67 @@
+import styled from 'styled-components';
+
+export const StyledTextArea = styled.textarea`
+  border: none;
+  outline: none;
+  width: 50%;
+  height: auto;
+  border-bottom-left-radius: 50px;
+  resize: none;  
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
```
Comment (drews-believer) [inline src/components/App/Randomizer/StyledRandomizerComponents.tsx]: why?) line № 10

File: src/components/App/Randomizer/Randomizer.tsx [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+export default function Randomizer() {
+  const [value, setValue] = React.useState('Your result will be here');
+  const [result, setResult] = React.useState<ResultInterface[]>([]);
+  const [inputValue, setInputValue]= React.useState(1);
+  
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value);
+  };
+  //TYPE IS ANY DUE TO MATERIAL UI DOCUMENTATION ISSUES
+  const handleInputChange = (event: any) => {
```
Comment (drews-believer) [inline src/components/App/Randomizer/Randomizer.tsx]: Remove any from event pls)

File: src/components/App/Randomizer/Randomizer.tsx [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+    setInputValue(Number(event.target.value));
+  };
+
+  let resultArray: string[] = [];
+  const randomize = () => {
+    resultArray = value.split('\n');
+    if(resultArray.length>=inputValue){
```
Comment (drews-believer) [inline src/components/App/Randomizer/Randomizer.tsx]: Where are spaces?) use prettier pls, when u are finished.

**Review by drews-believer — CHANGES_REQUESTED** [review]

**Review by yanosh-almashi — APPROVED** [review]

File: src/components/App/Randomizer/Randomizer.tsx (hunk 13 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: changed

File: src/components/App/Randomizer/StyledRandomizerComponents.tsx (hunk 12 above)
Comment (CBond05) [reply src/components/App/Randomizer/StyledRandomizerComponents.tsx]: fix

File: src/components/App/Randomizer/Randomizer.tsx (hunk 14 above)
Comment (CBond05) [reply src/components/App/Randomizer/Randomizer.tsx]: added

**Pull request closed without merging**